semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/Scalalib.scala

```diff
@@ -37,18 +37,23 @@ object Scalalib {
       builtinMethod("AnyRef", List(p.FINAL), "eq", Nil, List("that" -> "scala.AnyRef#"), "scala.Boolean#"),
       builtinMethod("AnyRef", List(p.FINAL), "ne", Nil, List("that" -> "scala.AnyRef#"), "scala.Boolean#"),
       builtinMethod("AnyRef", List(p.FINAL), "synchronized", List("T"), List("body" -> "scala.AnyRef.synchronized(T).[T]"), "scala.AnyRef.synchronized(T).[T]"))
-    builtinClass(Nil, "AnyRef", List("scala.Any#"), symbols.flatten)
+    builtin(k.CLASS, Nil, "AnyRef", List("scala.Any#"), symbols.flatten)
   }
 
   def nothingClass: ToplevelInfos = {
-    builtinClass(List(p.ABSTRACT, p.FINAL), "Nothing", List("scala.Any#"), Nil)
+    builtin(k.CLASS, List(p.ABSTRACT, p.FINAL), "Nothing", List("scala.Any#"), Nil)
   }
 
   def nullClass: ToplevelInfos = {
-    builtinClass(List(p.ABSTRACT, p.FINAL), "Null", List("scala.AnyRef#"), Nil)
+    builtin(k.CLASS, List(p.ABSTRACT, p.FINAL), "Null", List("scala.AnyRef#"), Nil)
   }
 
-  private def builtinClass(
+  def singletonTrait: ToplevelInfos = {
+    builtin(k.TRAIT, List(p.FINAL), "Singleton", List("scala.Any#"), Nil)
+  }
+
+  private def builtin(
+      kind: s.SymbolInformation.Kind,
```

Comment:
I agree union type would be nice but I can live with limiting it to SymbolInformation.Kind, which is a medium sized enum